extensions/rich_text_components/Svgdiagram/protractor.js

```diff
@@ -122,9 +128,15 @@ var customizeComponent = async function(modal, shapes, altText) {
     await waitFor.elementToBeClickable(
       shapeTool,
       'Could not click on the required tool');
-    var customTools = ['bezier', 'piechart'];
+    var customTools = ['bezier', 'piechart', 'svgupload'];
     if (customTools.includes(shapes[i])) {
       await shapeTool.click();
+      if (shapes[i] === 'svgupload') {
+        var imageUploadInput = element(
+          by.css('.protractor-test-photo-upload-input'));
+        absPath = path.resolve(__dirname, './circle.svg');
+        await imageUploadInput.sendKeys(absPath);
```

Comment:
Please make sure imageUploadInput has loaded before interacting with it. You can use `protractor_utils/actions.js`, which has a helper function that handles this for you

Reply:
@U8NWXD This file input element is invisible in the first place so I think that visibility check will not work on the element. This check has not been done previously also like in ```workflow.js``` uploadImage Function.

Reply:
@rohitkatlaa in that case you can use a waitFor statement, maybe waitFor.presenceOf, to check that the element has loaded.

Also, you are correct that some of the existing e2e code does not follow our current best practices. We are working on migrating it, but we haven't finished yet. In the meantime, since the e2e tests are already so flaky, we want to make sure that any new code is written as well as possible to avoid making things any worse. Not checking that elements have loaded before interacting with them is also one of the most common problems we see on PRs. We have listed these problems [here](https://github.com/oppia/oppia/wiki/End-to-End-Tests#codeowner-checks) on the wiki to be more clear about what we check in codeowner reviews

Reply:
Done.

Reply:
Thanks! Would you mind making the error message more specific? We don't always get line numbers in the stack traces, so the error message needs to be enough to debug the issue. Maybe `Image upload input element took too long to load`?

Reply:
Done.